stl/inc/regex

```diff
@@ -1725,6 +1721,13 @@ public:
             _Char_class_d = _Lookup_char_class(static_cast<_Elem>('D'));
         }
 
+        _Input_length = _STD distance(_Pfirst, _Plast);
+        if (_Input_length < 0) {
+            _Input_length = 0;
+        }

```

Comment:
Some caution is often good, but this is overly paranoid. We've already performed `_Adl_verify_range(_Pfirst, _Plast);` to rule out transposed ranges (in debug mode). If the iterators are transposed, we have bigger problems.